etl/src/v2/replication/apply.rs

```diff
@@ -321,6 +326,25 @@ where
                         false
                     )
                     .await?;
+
+                // If the apply loop is not in the middle of processing a transaction, call the hook's
+                // process_syncing_tables method so that apply worker:
+                //
+                // * Marks any table sync workers in sync wait state as catchup
+                // * Marks any sync done table sync workers as ready
+                //
+                // and the table sync worker:
+                //
+                // * Marks itself as sync done if it completes its catch phase
+                //
+                // This is done here as well in addition to at a commit boundary because we do not want
+                // the table sync workers to get stuck if there are no changes in the cdc stream.
+                if !state.handling_transaction() {
+                    let continue_loop = hook.process_syncing_tables(state.next_status_update.flush_lsn).await?;
```

Comment:
This is a good addition, but we might need to see how it performs because it might stall events processing if we end up in the worst case where we kickstart this and right after a new batch is received.